sqlmesh/core/dialect.py

```diff
@@ -795,38 +795,40 @@ def pandas_to_sql(
     )
 
 
-def set_default_schema_and_catalog(
+def set_default_catalog(
     table: str | exp.Table,
-    default_schema: t.Optional[str],
     default_catalog: t.Optional[str],
 ) -> exp.Table:
     table = exp.to_table(table)
 
-    if default_schema and not table.db:
-        table.set("db", exp.to_identifier(default_schema))
-    if default_catalog and not table.catalog:
-        table.set("catalog", exp.to_identifier(default_catalog))
+    if default_catalog and not table.catalog and table.db:
+        table.set("catalog", exp.parse_identifier(default_catalog))
 
     return table
 
 
 def normalize_model_name(
     table: str | exp.Table | exp.Column,
+    default_catalog: t.Optional[str],
     dialect: DialectType = None,
-    column_is_table: bool = False,
 ) -> str:
     if isinstance(table, exp.Column):
-        if column_is_table:
-            table = exp.table_(table.this, db=table.args.get("table"), catalog=table.args.get("db"))
-        else:
-            table = exp.table_(*reversed(table.parts[:-1]))  # type: ignore
+        table = exp.table_(table.this, db=table.args.get("table"), catalog=table.args.get("db"))
     else:
         table = exp.to_table(table, dialect=dialect)
 
-    return exp.table_name(normalize_identifiers(table, dialect=dialect))
+    table = set_default_catalog(table, default_catalog)
+    # An alternative way to do this is the following: exp.table_name(table, dialect=dialect, identify=True)
+    # This though would result in the names being normalized to the dialect AND the quotes while the below
```

Comment:
What does it mean to be "normalized to the quotes"?

Reply:
Updated to clarify that we use sqlglot dialect for the quotes instead of the dialect's quotes. If this doesn't help it might be useful to talk it through and figure out the english translation of this:
```
# An alternative way to do this is the following: exp.table_name(table, dialect=dialect, identify=True)
    # This though would result in the names being normalized to the target dialect AND the quotes while the below
    # approach just normalizes the names.
    # By just normalizing names and using sqlglot dialect for quotes this makes it easier for dialects that have
    # compatible normalization strategies but incompatible quoting to still work together without user hassle
```